### PR TITLE
Generate PerfType objects automatically based on Speed and Variant lists

### PR DESCRIPTION
**Pull request by msomatt**

### CODE REVIEW

File: app/views/tournament/leaderboard.scala [hunk 1]
```diff
@@ -79,11 +79,11 @@ object leaderboard {
           h1("Tournament winners"),
           div(cls := "tournament-leaderboards")(
             eliteWinners,
-            freqWinners(winners.hyperbullet, PerfType.Bullet, "HyperBullet"),
-            freqWinners(winners.bullet, PerfType.Bullet, "Bullet"),
```
Comment (lakinwecker) [inline app/views/tournament/leaderboard.scala:83]: You also changed the text "Bullet" vs "bullet"

**Comment by lakinwecker** [comment]
TL;DR: I'm fine with this being merged, provided we make a ticket about using a type-provider macro to re-instate the strongly typed versions of these in the long run: https://docs.scala-lang.org/overviews/macros/typeproviders.html#public-type-providers

Overall, I believe that we needed to do this sort of restructuring in order to more easily be able to add more ratings into the system whenever we add a new variant. From this perspective this is a positive and wanted change.  It does, however, reduce the type safety of the code overall, which is what you had noted in our discussion.  During my review I looked for places where this would have removed wanted type checks or exhaustivity checks, and I did not see any places where the exhaustivity checks would be reduced.. It seems that the code was already designed to allow for us to be incomplete in our dealings with ratings.  I can't remember, and I didn't have time to look into whether the lack of exhaustivity checks was due to lichess or us.  However, it does introduce the possibility that we'll mistype a string and won't know except for the bugs that it might cause.  For this reason, I think we consider re-instating the types via the above suggested typeproviders.  I think this is an excellent exercise in learning Scala as well.  It doesn't need to be done right away, I think we can make a task for it and then get to it in time.

File: app/views/tournament/leaderboard.scala (hunk 1 above)
Comment (msomatt) [reply app/views/tournament/leaderboard.scala:83]: Ah yeah good spot!
